### DISCUSSION TITLE
Branding: Adjust the ISSUER string for 2FA

### DISCUSSION THREAD

**Discussion by the-voidl** [zitadel/zitadel#5453] [Ideas]
The QR code you need for the initial setup of TOTP contains a string that identifies the Issuer of your account.

By default these QR Codes contain the string ZITADEL.
We would like to set this to our own company name since none of our users knows what that "ZITADEL" in their TOTP app stands for.

The generated string looks like this so far:
`otpauth://totp/ZITADEL:user@example.com?algorithm=SHA1&digits=6&issuer=ZITADEL&period=30&secret=AD7NX22ZGJJIEVAFMJOVA6H5XRJO4A79`

**Comment by fforootd** [comment]
Well that makes total sense 😁 since we allow to configure WebAuthN as well it is something we should make available

What do you think @hifabienne

**Comment by hifabienne** [comment]
Yes, and there is already an issue for this. Unfortunately the priorisation was never as high to make it to the next sprint.
https://github.com/zitadel/zitadel/issues/4712

**Reply by the-voidl** [reply]
This would have been great to be included in recently released https://github.com/zitadel/zitadel/issues/4607 due to the same category.

It would be nice to see that implemented in the near future :)

**Reply by hifabienne** [reply]
This has been released with 2.26  and will be deployed to zitade.cloud soon.

**Reply by ckhatton-pvfscaffolding** [reply]
I imagine this is for API and custom login portal, but how does one change the issuer via the console?

**Reply by rajatcing** [reply]
hey @ckhatton-pvfscaffolding at present, changing the issuer in the TOTP QR code directly from the zitadel console isn't supported.

**Reply by ckhatton-pvfscaffolding** [reply]
Ar ok! Thanks for replying. I tried setting the `ZITADEL_SYSTEMDEFAULTS_MULTIFACTORS_OTP_ISSUER` env to 'Example\ Name' during setup and adding it to the Linux zitadel.service, but that didn't seem to change it.

Example:
```
sudo ZITADEL_SYSTEMDEFAULTS_MULTIFACTORS_OTP_ISSUER=Example\ Name /usr/local/bin/zitadel setup --masterkeyFile /home/zitadel/zitadel-masterkey
```

Do you know why?

**Reply by ckhatton-pvfscaffolding** [reply]
Conversion moved to [Discord](https://discord.com/channels/927474939156643850/1430201017546571826).

**Reply by timarney** [reply]
Is this available to set somewhere -- env var or admin console ?

I tried the Discord link but it didn't load

**Reply by maroquio** [reply]
Hi, folks. I think this is an important issue to be solved. The ZITADEL name in the Authenticator app says nothing to a SaaS user. I hope you fix it soon.

**Reply by ckhatton-pvfscaffolding** [reply]
@timarney, not much occurred on the Discord thread, I was just told to upgrade to the latest version of Zitadel and try pass the ENV again via the setup command - I haven't had a chance to try it again.

**Reply by zacharya19** [reply]
FYI: Using zitadel `4.15.0` with `ZITADEL_SYSTEMDEFAULTS_MULTIFACTORS_OTP_ISSUER` env var worked for me
Setting it using the helm values didn't work:
```
SystemDefaults:
  Multifactors:
    Issuer: "Example"
```

**Reply by m-schubert** [reply]
@zacharya19, the config values you specified are missing the `OTP` key in the tree. It should be:

```
SystemDefaults:
  Multifactors:
    OTP:
      Issuer: "Example"
```

I just tested and it worked for me!
